activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -119,6 +126,10 @@ def initialize(connection, logger = nil, config = {}) # :nodoc:
         else
           @prepared_statements = false
         end
+
+        @advisory_locks = self.class.type_cast_config_to_boolean(
```

Comment:
I know it's a departure from `@prepared_statements`, but I suspect this would be clearer as `@advisory_locks_enabled`, particularly for the callers below.